examples/shader/blackhole.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build ignore
+// +build ignore
+
+package main
+
+var Time float
+var ScreenSize vec2
+var Cursor vec2
+
+func Fragment(position vec4, texCoord vec2, color vec4) vec4 {
+	uv := position.xy / ScreenSize.xy
+	texC := uv
```

Comment:
This texC is not used anywhere.

Reply:
It is used as texture coordinates in `imageSrcNAt` a few lines below

Reply:
texC is reassigned before passing it to `imageSrcNAt`

Reply:
Oh you meant like that, yeah uv could also go away I believe then (more like `uv` is only used with `texC`)
I think my block from below comment might cover this

Reply:
> texC = mix(texC2, (texC*2.0-lpos*2.0)*0.2*0.5+lpos, (1.0 / (distance((texC2*2.0-lpos*2.0)*10.0*0.5+lpos, lpos) - 1.0)))

Ah, I found that `texC` is used in the right side, but is this intended?

Reply:
Using the same variable name for multiple contexts is confusing. Let's use a different variable names.

Reply:
Yeah it was intended because the whole shader is a distortion of texture coordinates, and there was no extra need for an additional variable (also it's translated from shadertoy)

I agree additional variables might help describe the different steps better, but I wouldn't know how to name it :/
Maybe keep uv for the first transformations, and declare `texC = mix(` later using `uv` you mean ?